tokio/src/runtime/task/list.rs

```diff
@@ -179,26 +210,53 @@ impl<S: 'static> OwnedTasks<S> {
 
         // safety: We just checked that the provided task is not in some other
         // linked list.
-        unsafe { self.inner.lock().list.remove(task.header_ptr()) }
+        unsafe { self.remove_inner(task) }
+    }
+
+    #[inline]
+    unsafe fn remove_inner(&self, task: &Task<S>) -> Option<Task<S>> {
+        // Safety: it is safe, because every task has one task_id
+        let task_id = unsafe { Header::get_id(task.header_ptr()) };
+        let mut lock = self.segment_inner(task_id.0 as usize);
+        let task = lock.remove(task.header_ptr())?;
+        drop(lock);
+        self.count.fetch_sub(1, Ordering::Relaxed);
```

Comment:
You should only subtract here if `task.is_some()`.

Reply:
I have used the `?` operator to quickly return. So if the task is `None`, we will not subtract here.